### PR TITLE
Updated style-guide - Added rules for variable declaration

### PR DESCRIPTION
**Pull request by Ruchirchheda**
Added rules for auto, const reference, unsigned

### CODE REVIEW

File: doc/advanced/content/pcl_style_guide.rst [hunk 1]
```diff
@@ -410,3 +410,16 @@ For the compute, filter, segment, etc. type methods the following rules apply:
...
+^^^^^^^^^^^^^^^^^^^^^^^
+
+3.3.1 Use of auto
+"""""""""""""""""
+* For Iterators auto must be used as much as possible 
+* In all the other cases auto can be used at the author's discretion
+
+3.3.2 Type qualifiers of variables
+""""""""""""""""""""""""""""""""""
+* Use of const reference variables if the value is not being modified
```
Comment (SergioRAgostinho) [inline doc/advanced/content/pcl_style_guide.rst:423]:
```suggestion
* Declare variables const when they don't need to be modified.
* Use const references whenever you don't need a copy of the variable. 
```

File: doc/advanced/content/pcl_style_guide.rst [hunk 2]
```diff
@@ -410,3 +410,16 @@ For the compute, filter, segment, etc. type methods the following rules apply:
...
 * The output arguments will always be passed by reference.
 
+3.3. Object declaration
+^^^^^^^^^^^^^^^^^^^^^^^
+
+3.3.1 Use of auto
+"""""""""""""""""
+* For Iterators auto must be used as much as possible 
+* In all the other cases auto can be used at the author's discretion
+
```
Comment (SergioRAgostinho) [inline doc/advanced/content/pcl_style_guide.rst:420]:
```suggestion
* Use const auto references by default in range loops. Drop the const if the item needs to be modified.

```

**Comment by Morwenn** [comment]
I thought that `auto&&` was almost always the favoured one in foreach loops because it plays nice even with pseudo-reference types à la `std::vector<bool>::reference`.

**Comment by SergioRAgostinho** [comment]
I don't believe taking forward references by default is the safe choice here. Users which really need it will definitely know when to use it. The others, I'd rather keep them on the const ref by default.

**Pull request merged**